### PR TITLE
Store Alerts: Update widths to match main body

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Partial fix for https://github.com/woocommerce/woocommerce/issues/47034

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Launch a new JN site and skip profiler
2. SSH into the site and force a store alert to appear by running `wp option set woocommerce_db_version 8.9.0`
3. Visit wc-admin pages and legacy pages to see the store alert widths appropriate. Be sure to vary the window size to make sure its responsive.
4. Now set the homescreen view to two columns  by hiding the setup list and refreshing the page. The width of the alert should match the two columns.

Tip: Use `wp option set woocommerce_default_homepage_layout single_column` to reset to single column

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Update Store Alert widths to match main body

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Review by chihsuan — COMMENTED** [review]
Thanks for working on this! @psealock  It tested well when set the homescreen view to two columns by hiding the setup list. 

However, when I changed layout via display options in the homescreen, the alert width did not match the two columns. I've attached a screenshot for reference. Can you please take a look? 🙏 

Screenshot 2024-06-19 at 13 35 39

It's not related to the PR but it looks like the buttons are too close in mobile view. I think it's better to add some padding between them.

Screenshot 2024-06-19 at 13 34 17

**Review by moon0326 — COMMENTED** [review]
Other than what Chi mentioned already, LGTM!

**Comment by psealock** [comment]
@moon0326 @chihsuan,

In order to get the layout to respect userPreferences, I needed to convert `<Layout />` from a Class component to a functional one. This required a sizeable refactor, so this PR will now require testing around the Alert behaviour.

1. Trigger multiple notes. First see what's already in your DB.
```sql
SELECT note_id,is_read,is_deleted,status,type from wp_wc_admin_notes WHERE type = 'update'
```
2. Make a few "unactioned".
```sql
UPDATE wp_wc_admin_notes SET status = 'unactioned' WHERE note_id IN (1,2,3,4);
```
3. You may need to mark some as not deleted.
```sql 
UPDATE wp_wc_admin_notes SET is_deleted = 0 WHERE note_id IN (1,2,3,4);
```
4. Reload the page, you should have a few Alerts visible. Do the buttons work? Does the "X" work? Can you toggle through them?

Now test the updated behaviour:
1. Is the layout setting respected?

image

2. Check the button spacing at narrow widths? Do they stack as you'd expect?

**Review by chihsuan — COMMENTED** [review]
Thanks for the update!  @psealock 

> Do the buttons work? Does the "X" work? Can you toggle through them? 

✅ 

> Is the layout setting respected? 

✅ 
> Check the button spacing at narrow widths? Do they stack as you'd expect? 

✅ 

Do we also need to update the widths to match the main body for other pages? I'm seeing the alert width looks weird in the other pages. 🤔 

Screenshot 2024-06-26 at 12 13 31



Nitpick: When using a column view, the pagination button width is not consistent when navigating through the alerts and causes the UI to jump. 

https://github.com/woocommerce/woocommerce/assets/4344253/70299ce9-1e43-42f7-a853-19a6551e819c

**Comment by psealock** [comment]
Good catch @chihsuan, in https://github.com/woocommerce/woocommerce/pull/48487/commits/103d68ef408398e59f22f2f0eba3befbc6455e27 I handle this case where its single column layout but in a wc-admin page not the homescreen.

>Nitpick: When using a column view, the pagination button width is not consistent when navigating through the alerts and causes the UI to jump.

Yup, thats noted! I'll handle the styling of the inside of the note in a follow up to finish up https://github.com/woocommerce/woocommerce/issues/47034

**Comment by chihsuan** [comment]
> Good catch @chihsuan, in [103d68e](https://github.com/woocommerce/woocommerce/commit/103d68ef408398e59f22f2f0eba3befbc6455e27) I handle this case where its single column layout but in a wc-admin page not the homescreen.

Thanks! It looks better 🙂. But I found the alert width still does not match the main body for large screens because of the max-width constraint. Should we only apply the max-width constraint for Home screen alerts?

Screenshot 2024-06-26 at 14 12 18

**Review by moon0326 — APPROVED** [review]
The new changes have fixed all the issues @chihsuan mentioned 👍 

LGTM and tested well 🚀

**Comment by psealock** [comment]
>Should we only apply the max-width constraint for Home screen alerts?

Good question @chihsuan, I never considered the aim was to keep it as wide as the content for really wide screens. @elizaan36 I have a max-width of 1032px for all non-Homescreen screens. Do you think it should be allowed to fully go across the page? I'm leaning towards keeping the max-width, but another opinion would be helpful 🙇🏽‍♂️

**Comment by elizaan36** [comment]
> @elizaan36 I have a max-width of 1032px for all non-Homescreen screens. Do you think it should be allowed to fully go across the page? I'm leaning towards keeping the max-width, but another opinion would be helpfu

The notice should ideally match the width of the content on the page to feel more integrated, so yes I think it should be allowed to fully go across the page to align with the max width of the overall container.

**Comment by psealock** [comment]
@chihsuan, I think I finally got it 😄 Can you test the following and be sure to check responsive sizes too?

1. Homescreen single column
2. Homescreen double column
3. Settings page
4. Customers page

🙇🏽‍♂️

File: plugins/woocommerce-admin/client/layout/store-alerts/style.scss [hunk 1]
```diff
@@ -62,12 +104,22 @@
...
-			margin-top: $gap-large;
-		}
+		// left: 0;
+		// width: 100%;
+		// max-width: 1032px;
+
+		// @include breakpoint("<782px") {
+		// 	left: 0;
+		// 	margin-top: $gap-large;
+		// }
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/layout/store-alerts/style.scss]: Can we remove these lines? 😃

**Review by chihsuan — APPROVED** [review]
Looks perfect, thanks!  🎉

Screenshot 2024-06-27 at 11 42 06

Screenshot 2024-06-27 at 11 41 56

**Comment by tomalec** [comment]
As for reference, this PR introduced an issue which we hot fixed in CFE https://github.com/woocommerce/woocommerce/issues/53017#issuecomment-2526297520

**Pull request merged**